pkg/controller/perconaservermongodb/pbm.go

```diff
@@ -236,6 +292,15 @@ func (r *ReconcilePerconaServerMongoDB) reconcilePiTRConfig(ctx context.Context,
 	defer pbm.Close(ctx)
 
 	if err := enablePiTRIfNeeded(ctx, pbm, cr); err != nil {
+		if backup.IsErrNoDocuments(err) {
+			if cr.CompareVersion("1.20.0") < 0 && len(cr.Spec.Backup.Storages) == 1 {
```

Comment:
Instead of checking if the storages are equal to 1 here, I would rather push this responsibility inside the reconcile function and do something like this

```
	if len(cr.Spec.Backup.Storages) != 1 {
		log.Info("Expected exactly one storage for PiTR in legacy version", "configured", len(storages))
		return nil
	}
```

This makes the logic inside the reconcile function that selects the 1st storage it finds much easier to follow on why it happens.